front/app/components/Form/Components/Layouts/CLPageLayout.tsx

```diff
@@ -149,6 +149,38 @@ const CLPageLayout = memo(
       }
     };
 
+    const handlePrevious = () => {
+      const currentPageCategorization = uiPages[currentStep];
+      // Get scopes of elements with rules on the current page
+      const ruleElementsScopes = currentPageCategorization.elements
+        .filter((element) => {
+          return element.options?.hasRule;
+        })
+        .flatMap((filteredElement) => {
+          const elementScope = filteredElement.scope?.split('/').pop();
+          return elementScope || '';
+        });
+
+      const dataWithoutRuleValues = Object.fromEntries(
```

Comment:
Think this is the case, but just to be sure this only removes the data for rules on the current page?